_posts/2021-07-28-solid-principles-illustrated-with-python.md

```diff
@@ -0,0 +1,289 @@
+# S.O.L.I.D. principles illustrated in Python
+
+
+SOLID is an acronym created by Bob Martin and Michael Feathers that refers to 
+five fundamental principles that help engineers write maintainable code.
+
+## S: Single Responsibility principle SRP
+
+"A module, class, function should have one and only one reason to change."
+
+We say that they have "high cohesion" so that we can reuse them later on
+without side effect.
+
+For example, let us say we are dealing with an inventory management system
+that is going to track the stock per product.
+
+```python
+class Product:
+    def __init__(self, sku, stock):
+        self.sku = sku
+        self.stock = stock
```

Comment:
i think this example is too simple, as it is it would be crazy to split it.  of course it's just meant to be a simple example but still if you add a few more attributes it will feel more realistic, and the decision to split will make more sense.

```suggestion
class Product:
    def __init__(self, sku, stock):
        self.sku = sku
        self.price = price
        self.colour = colour
        self.current_stock = stock
        self.minimum_stock = ...
        self.maxium_stock = ...
```

ps hi!  :wave:

Reply:
Hi Harry, such a nice surprise to see your comment here :) Thank you!
I just did a presentation today about how we were using the Dependency Injection Pattern at made, following up on this.
You are right, my example is too simple. Re-reading [your book](https://www.cosmicpython.com/), I saw that it makes sense to have a whole business case behind it to explain the principles. I should elaborate more on the business case to make more sense.
I wanted it to be really simple and I think I will carry on with the Inventory Management System, but I should give more details.
Good to have some feedback, thanks!